### PR TITLE
Added Linux LVM support using libbfio and libvslvm #1148

### PR DESCRIPTION
**Pull request by joachimmetz**

### CODE REVIEW

**Comment by joachimmetz** [comment]
@bcarrier PTAL per separate mail conversion. I'll need to some support to get this cross-platform and inline with design ideas behind the pool layer seeing there is very little developers information that I was able to find.

**Comment by joachimmetz** [comment]
@bcarrier any help from the maintainers side? The documentation for the pool layer is mostly missing https://github.com/sleuthkit/sleuthkit/issues/2748

File: tsk/pool/pool_open.cpp [hunk 1]
```diff
@@ -3,16 +3,19 @@
  *
  * Brian Carrier [carrier <at> sleuthkit [dot] org]
  * Copyright (c) 2018-2019 BlackBag Technologies.  All Rights reserved
+ * Copyright (c) 2022 Joachim Metz <joachim.metz@gmail.com>
  *
  * This software is distributed under the Common Public License 1.0
  */
-#include "../fs/tsk_apfs.hpp"
+#include "tsk/tsk_config.h"
```
Comment (APriestman) [inline tsk/pool/pool_open.cpp:10]: This breaks the Visual Studio build since the file is not found.

Comment (joachimmetz) [reply tsk/pool/pool_open.cpp:10]: what is not found? can't this be addressed with an additional include path in the VS configuration?

Comment (APriestman) [reply tsk/pool/pool_open.cpp:10]: tsk_config.h does not exist.

Comment (joachimmetz) [reply tsk/pool/pool_open.cpp:10]: ah so it is not generated for VS, any conditional imports I should use? what is the recommended way of dealing with this in this codebase?

Comment (APriestman) [reply tsk/pool/pool_open.cpp:10]: You could try importing tsk_base_i.h instead - it has a check on HAVE_CONFIG_H before trying to include it.

Comment (joachimmetz) [reply tsk/pool/pool_open.cpp:10]: I'll have a look at that (tomorrow local time), the VS configuration has to define HAVE_LIBBFIO and HAVE_LIBVSLVM and ideally LIBVSLVM_HAVE_BFIO as well

Comment (joachimmetz) [reply tsk/pool/pool_open.cpp]: Made changes to headers

**Review by APriestman — APPROVED** [review]

**Comment by typeryougishiki** [comment]
Can I use this supply to get fs info from lvm volume？

**Comment by markmckinnon** [comment]
As long as you are willing to build sleuthkit from source on Linux and make the proper libraries available you can see that information

**Comment by leean518** [comment]
Does LVM support not work when developing on the windows operating systems?

**Comment by leean518** [comment]
@joachimmetz I believe I did. But just to confirm, can you specify how to compile with LVM support on Windows?

**Comment by joachimmetz** [comment]
> But just to confirm, can you specify how to compile with LVM support on Windows?

do you see DLLs for libvslvm and libbfio ? if not you likely did not compile with LVM support. I don't maintain the Visual Studio configuration of this project, so I don't know.

**Comment by leean518** [comment]
> 

I had to add the libraries myself as they are not included. The main problem I am having is how to integrate it with the sleuth kit so that the library is able to detect where it is.

**Pull request merged**